node/modules/storageminer_idxprov.go

```diff
@@ -116,3 +131,155 @@ func IndexProvider(cfg config.IndexProviderConfig) func(params IdxProv, marketHo
+	return datatransferv2.ChannelID{}, fmt.Errorf("not implemented")
+}
+
+func (i *indexerDT) OpenPullDataChannel(ctx context.Context, to peer.ID, voucher datatransferv2.TypedVoucher, baseCid cid.Cid, selector datamodel.Node, options ...datatransferv2.TransferOption) (datatransferv2.ChannelID, error) {
+	return datatransferv2.ChannelID{}, fmt.Errorf("not implemented")
+}
+
+func (i *indexerDT) SendVoucher(ctx context.Context, chid datatransferv2.ChannelID, voucher datatransferv2.TypedVoucher) error {
+	return fmt.Errorf("not implemented")
+}
+
+func (i *indexerDT) SendVoucherResult(ctx context.Context, chid datatransferv2.ChannelID, voucherResult datatransferv2.TypedVoucher) error {
+	return fmt.Errorf("not implemented")
+}
+
+func (i *indexerDT) UpdateValidationStatus(ctx context.Context, chid datatransferv2.ChannelID, validationResult datatransferv2.ValidationResult) error {
+	return fmt.Errorf("not implemented")
+}
+
+func (i *indexerDT) CloseDataTransferChannel(ctx context.Context, chid datatransferv2.ChannelID) error {
+	return fmt.Errorf("not implemented")
+}
+
+func (i *indexerDT) PauseDataTransferChannel(ctx context.Context, chid datatransferv2.ChannelID) error {
+	return fmt.Errorf("not implemented")
+}
+
+func (i *indexerDT) ResumeDataTransferChannel(ctx context.Context, chid datatransferv2.ChannelID) error {
+	return fmt.Errorf("not implemented")
+}
+
+func (i *indexerDT) TransferChannelStatus(ctx context.Context, x datatransferv2.ChannelID) datatransferv2.Status {
+	return 0
+}
+
+func (i *indexerDT) ChannelState(ctx context.Context, chid datatransferv2.ChannelID) (datatransferv2.ChannelState, error) {
+	return nil, fmt.Errorf("not implemented")
+}
+
+func (i *indexerDT) SubscribeToEvents(subscriber datatransferv2.Subscriber) datatransferv2.Unsubscribe {
+	return func() {}
+}
+
+func (i *indexerDT) InProgressChannels(ctx context.Context) (map[datatransferv2.ChannelID]datatransferv2.ChannelState, error) {
+	return nil, fmt.Errorf("not implemented")
+}
+
+func (i *indexerDT) RestartDataTransferChannel(ctx context.Context, chid datatransferv2.ChannelID) error {
+	return fmt.Errorf("not implemented")
+}
+
+type dtv1Voucher struct {
+	t string
+}
+
+func (d *dtv1Voucher) Type() datatransfer.TypeIdentifier {
+	return datatransfer.TypeIdentifier(d.t)
+}
+
+type dtv1VoucherResult struct {
+	t string
+}
+
+func (d *dtv1VoucherResult) Type() datatransfer.TypeIdentifier {
+	return datatransfer.TypeIdentifier(d.t)
+}
+
+type dtv1ReqValidator struct {
+	v datatransferv2.RequestValidator
+}
+
+func (d *dtv1ReqValidator) ValidatePush(isRestart bool, chid datatransfer.ChannelID, sender peer.ID, voucher datatransfer.Voucher, baseCid cid.Cid, selector ipld.Node) (datatransfer.VoucherResult, error) {
+	d2v := types.BindnodeRegistry.TypeToNode(voucher.Type())
```

Comment:
This is not going to work. You're losing your internal data here, and you're using a BindnodeRegistry that will not recognize the value. I believe what you want is to cast this to your legs1voucher type. Then call .AsVoucher().Voucher

so:

```golang
d2v := voucher.(*legsVoucherDTv1).AsVoucher().Voucher
```

Reply:
alternatively:
```golang
d2v := dtsync.BindnodeRegistry.TypeToNode(voucher.(*legsVoucherDTv1).Voucher)
```